parmed/amber/parameters.py

```diff
@@ -200,13 +211,22 @@ def __init__(self, *filenames):
         self.residues = dict()
         for filename in filenames:
             if isinstance(filename, string_types):
-                self.load_parameters(filename)
+                if self.id_format(filename): # dat or frcmod file
```

Comment:
I would do it the other way around.  Change this to:

```Python
if AmberOFFLibrary.id_format(filename):
    self.residues.update(AmberOFFLibrary.parse(filename)
else:
    self.load_parameters(filename)
```

The reason is that if you pass a wrong file type to it, then it will error as though it's being treated as an OFF library, which is a bit misleading.  Since we're special-casing AmberOFFLibrary, we should handle *that* case explicitly and pass everything else to the core parser.